x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -360,12 +364,27 @@ ResolvedIndices resolveIndicesAndAliases(
                             nameExpressionResolver,
                             indicesRequest.includeDataStreams()
                         )) {
-                            resolvedIndicesBuilder.addLocal(
+                            localExpressions.add(
                                 IndexNameExpressionResolver.combineSelectorExpression(authorizedIndex, allIndicesPatternSelector)
                             );
                         }
                     }
                 }
+
+                var resolvedExpressionsBuilder = ResolvedIndexExpressions.builder();
+                resolvedExpressionsBuilder.addExpressions(
+                    Metadata.ALL,
```

Comment:
This is not necessarily the original expression which can also be `""` or with selectors, e.g. `_all::data`.